### PR TITLE
SNOW-1014048: Set `PARAM_APPLICATIONS` for snowpark ML usage tracking

### PR DESCRIPTION
**Pull request by sfc-gh-thoyt**
This fix enables tracking snowpark ML usage in sprocs, notebooks, and streamlit apps. This is a workaround because statement params telemetry is not received from sprocs.

* set "applications" parameter if snowpark ML is present
* fix helper method name
* Make sure "streamlit" is also set if present

Please answer these questions before submitting your pull requests. Thanks!

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 - Added support for an optional `date_part` argument in function `last_day`
 - `SessionBuilder.app_name` will set the query_tag after the session is created.
+- Set `"application"` parameter for snowpark ML usage tracking.
```
Comment (sfc-gh-yixie) [inline CHANGELOG.md:9]: This isn't needed because it's not related to users.

File: src/snowflake/snowpark/_internal/server_connection.py [hunk 2]
```diff
@@ -710,7 +718,10 @@ def _fix_pandas_df_fixed_type(
                 # we try to strictly use astype("int64") in this scenario. If the values are too large to
                 # fit in int64, an OverflowError is thrown and we rely on to_numeric to choose and appropriate
                 # floating datatype to represent the number.
-                if column_metadata.precision > 10 and not pd_df[pandas_col_name].hasnans:
+                if (
+                    column_metadata.precision > 10
+                    and not pd_df[pandas_col_name].hasnans
+                ):
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/_internal/server_connection.py:724]: Is this changed by a linting tool?

Comment (sfc-gh-thoyt) [reply src/snowflake/snowpark/_internal/server_connection.py:724]: Yes

**Review by sfc-gh-trichards — APPROVED** [review]
i'm pretty sure this will be totally fine, I messaged @sfc-gh-wgrus about it just in case

**Review by sfc-gh-yixie — APPROVED** [review]

**Pull request merged**